### PR TITLE
feat: support put & get & upload stage for FlightSQL

### PR DESCRIPTION
**Pull request by everpcpc**
* support put & get for `FlightSQL`
* support upload to stage for `FlightSQL`
* put & get use presigned url

### CODE REVIEW

File: core/src/client.rs [hunk 1]
```diff
@@ -421,29 +384,55 @@ impl APIClient {
...
+
+    pub async fn upload_to_stage(&self, stage: &str, data: Reader, size: u64) -> Result<()> {
         if self.presigned_url_disabled {
-            self.upload_to_stage_with_stream(stage_location, data, size)
-                .await
+            self.upload_to_stage_with_stream(stage, data, size).await
         } else {
-            self.upload_to_stage_with_presigned(stage_location, data, size)
-                .await
+            let presigned = self.get_presigned_upload_url(stage).await?;
```
Comment (b41sh) [inline core/src/client.rs:422]: For the storage that does not support the presign, can we use stream instead?

Comment (everpcpc) [reply core/src/client.rs:422]: We have a client argument `presigned_url_disabled` in DSN which can do this.

Comment (everpcpc) [reply core/src/client.rs:422]: BTW, this argument only works for HTTP API. FlightSQL will always do a presigned upload.

Comment (b41sh) [reply core/src/client.rs:422]: ok

**Pull request merged**